### ISSUE TITLE
Client: Not able to flip sort order

### ISSUE THREAD

**Issue by fdellwing** [osTicket/osTicket#3571]
Its not possible to sort columns by DESC in the client. Everytime you click on the column header to sort, it sorts ASC. In agent view it works fine. In v1.9.12 it works also for clients.

v1.10 from http://osticket.com/download

**Comment by fdellwing** [comment]
Did some investigation:

```php
$negorder=$order=='-'?'ASC':'DESC'; //Negate the sorting
```

At this point `$order` is always `-` (meaning DESC).

`$order` gets written here:

```php
if ($_REQUEST['order'] && $orderWays[strtoupper($_REQUEST['order'])])
    $order = $orderWays[strtoupper($_REQUEST['order'])];
else
    $order = $orderWays['DESC'];
```

So 
```php
$_REQUEST['order'] && $orderWays[strtoupper($_REQUEST['order'])]
```
seems to be always false.

That is the case, because `$orderWays['DESC'] == ''`

To fix this you could write the if like this:

```php
if ($_REQUEST['order'] && ($_REQUEST['order'] == 'ASC' || $orderWays[strtoupper($_REQUEST['order'])]))
```
Oh and please do the non english users a favour and revert the code after line 206!

It should look like this:

```html
    <thead>
        <tr>
            <th nowrap>
                <a href="tickets.php?sort=ID&order=<?php echo $negorder; ?><?php echo $qstr; ?>" title="<?php echo sprintf(__('Sort by %s %s'), __('Ticket #'), __($negorder)); ?>"><?php echo __('Ticket #');?></a>
            </th>
            <th width="120">
                <a href="tickets.php?sort=date&order=<?php echo $negorder; ?><?php echo $qstr; ?>" title="<?php echo sprintf(__('Sort by %s %s'), __('Create Date'), __($negorder)); ?>"><?php echo __('Create Date');?></a>
            </th>
            <th width="100">
                <a href="tickets.php?sort=status&order=<?php echo $negorder; ?><?php echo $qstr; ?>" title="<?php echo sprintf(__('Sort by %s %s'), __('Status'), __($negorder)); ?>"><?php echo __('Status');?></a>
            </th>
            <th width="320">
                <a href="tickets.php?sort=subj&order=<?php echo $negorder; ?><?php echo $qstr; ?>" title="<?php echo sprintf(__('Sort by %s %s'), __('Subject'), __($negorder)); ?>"><?php echo __('Subject');?></a>
            </th>
            <th width="120">
                <a href="tickets.php?sort=dept&order=<?php echo $negorder; ?><?php echo $qstr; ?>" title="<?php echo sprintf(__('Sort by %s %s'), __('Department'), __($negorder)); ?>"><?php echo __('Department');?></a>
            </th>
        </tr>
    </thead>
```
If you have a localization function, please use it!

**Comment by jdelhome3578** [comment]
I believe #3409 has already fixed this issue.

**Comment by fdellwing** [comment]
@jdelhome3578 Nope, has nothing to do with this problem. That PR is already merged and in this version. But it does not fix the $negorder problem.

This PR will fix it:  #3572
